plugins/search/src/components/SearchTypeFacet/SearchTypeFacet.stories.tsx

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { useState } from 'react';
+import CatalogIcon from '@material-ui/icons/MenuBook';
+import DocsIcon from '@material-ui/icons/Description';
+
+import { SearchTypeFacet } from '../index';
+import { SearchContext } from '../SearchContext';
+
+export default {
+  title: 'Plugins/Search/SearchTypeFacet',
+  component: SearchTypeFacet,
+};
+
+export const Default = () => {
+  const [types, setTypes] = useState<string[]>([]);
```

Comment:
is the adopter of the this component meant to define a state and state setter? Otherwise you could follow the same way we have been doing it for other components in storybook (see [example](https://backstage.io/storybook/?path=/story/plugins-search-searchbar--default)) by using a mock defaultValue for the context.

```tsx
const defaultValue = {
  types: [],
  setTypes: () => {},
};
```

and then 

```tsx
<SearchContext.Provider value={defaultValue}>
....
```

Reply:
Hmm...  I had copied this from another example.  I think without the state implementation, the story doesn't work interactively.  On the one you linked, the clear button doesn't actually clear the term, for example.

Maybe we should have a utility search context provider that's shared by all the storybook stories?

Reply:
Get it, and yes that is correct, the interactions doesn't work. I don't know how much you interact with the functionality in a story versus want to see the actual implementation. And using the SearchBar (that I gave as an example) it gets cleared under the good, so it's more "this is how you use the component" so that one can copy/paste it for usage. So just one thing to think about is - if additional state setters etc is added in the storybook example, the user can think of it as needed to use the component, but in this case that is not the case. But maybe a comment works too

Reply:
Refactored all of the search plugin stories a bit so that the stories are way more relevant to users (mostly by using `decorators` and providing a mocked search API to provide results). Check out: https://github.com/backstage/backstage/pull/8646/commits/dd95ac91ccfc4d6437ec2850051f444df20826ed